internal/handlers/msg_comparison_test.go

```diff
@@ -0,0 +1,345 @@
+package handlers
+
+import (
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+	"github.com/FerretDB/FerretDB/internal/util/testutil"
+	"github.com/stretchr/testify/assert"
+	"math"
+	"testing"
+	"time"
+)
+
+const (
+	DoubleDataType                 = "double"
+	DoubleZeroDataType             = "double-zero"
+	DoubleNegativeInfinityDataType = "double-negative-infinity"
+	DoublePositiveInfinityDataType = "double-positive-infinity"
+	ArrayDataType                  = "array"
+	EmptyArrayDataType             = "array-empty"
+	DoubleNanDataType              = "double-nan"
+	DoubleMaxDataType              = "double-max"
+	DoubleSmallestDataType         = "double-smallest"
+	StringDataType                 = "string"
+	EmptyStringDataType            = "string-empty"
+	DataDataType                   = "data"
+	BoolDataType                   = "bool"
+	DatetimeDataType               = "datetime"
+	Int32DataType                  = "int32"
+	Int32ZeroDataType              = "int32-zero"
+	Int32MaxDataType               = "int32-max"
+	Int32MinDataType               = "int32-min"
+	Int64DataType                  = "int64"
+	Int64ZeroDataType              = "int64-zero"
+	Int64MaxDataType               = "int64-max"
+	Int64MinDataType               = "int64-min"
+	TimestampDataType              = "timestamp"
+	SchemaWithAllTypes             = "values"
+	CollectionWithAllTypes         = "values"
+	BinaryDataType                 = "binary"
+)
+
+func TestComparison(t *testing.T) {
+	dataValues := map[string]any{
+		DoubleDataType:                 42.13,
+		DoubleNegativeInfinityDataType: math.Inf(-1),
+		DoublePositiveInfinityDataType: math.Inf(+1),
+		DoubleZeroDataType:             0.0,
+		DoubleNanDataType:              math.NaN(),
+		DoubleMaxDataType:              math.MaxFloat64,
+		DoubleSmallestDataType:         math.SmallestNonzeroFloat64,
+		StringDataType:                 "foo",
+		EmptyStringDataType:            "",
+		BoolDataType:                   true,
+		DataDataType:                   time.Date(2020, 2, 15, 9, 34, 33, 0, time.UTC).Local(),
+		DatetimeDataType:               time.Date(2021, 7, 27, 9, 35, 42, 123000000, time.UTC).Local(),
+		Int32DataType:                  int32(42),
+		Int32ZeroDataType:              int32(0),
+		Int32MaxDataType:               int32(2147483647),
+		Int32MinDataType:               int32(-2147483648),
+		Int64DataType:                  int64(42),
+		Int64ZeroDataType:              int64(0),
+		Int64MaxDataType:               int64(9223372036854775807),
+		Int64MinDataType:               int64(-9223372036854775808),
+		TimestampDataType:              types.Timestamp(42),
+		ArrayDataType:                  must.NotFail(types.NewArray("array", int32(42))),
+		EmptyArrayDataType:             must.NotFail(types.NewArray()),
+		BinaryDataType:                 types.Binary{Subtype: types.BinaryUser, B: []byte{42, 0, 13}},
+	}
```

Comment:
Why we need that map?

Reply:
I made a map that corresponds to the values of the table, as a convenience so that in each test case or in another test I don’t write the value manually.

Reply:
Let's remove both constants and this map: data is fixed by test_db, not them, and also replacing one constant with another does not achieve much.

Reply:
Agreed